### PR TITLE
fix(deps): upgrade wait-port to v1 and listen on correct address for framework

### LINKED ISSUE #4845: After upgrade to 10.11.1 netlify dev hangs on "waiting for port"
### Describe the bug

Upgraded to latest 10.11.1, running `netlify dev`
It just keeps hanging on "waiting for port forever"

⠋ Waiting for framework port 8080. This can be configured using the 'targetPort' property in the netlify.toml

No matter if I add the `targetPort` to netlify.toml or not, or also whhich port (tried 8080, 8888, 8081 etc)

```shell
valhalla-landers on  main [?] via  v18.6.0 on ☁️   took 45s
❯ netlify dev
(node:3821) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
wasm streaming compile failed: TypeError: Failed to parse URL from /usr/local/lib/node_modules/netlify-cli/node_modules/netlify-redirector/lib/redirects.wasm
falling back to ArrayBuffer instantiation
◈ Netlify Dev ◈
◈ Ignored general context env var: LANG (defined in process)
◈ Ignored general context env var: LC_ALL (defined in process)
◈ Injected .env file env var: ELEVENTY_ENV
◈ Injected .env file env var: SITE_ID
◈ Injected .env file env var: BROWSER
◈ Injected .env file env var: WCT_API_KEY
◈ Injected .env file env var: UPSTASH_REDIS_REST_URL
◈ Injected .env file env var: UPSTASH_REDIS_REST_TOKEN
◈ Injected .env file env var: UPSTASH_CLOAKING_REDIS_REST_URL
◈ Injected .env file env var: UPSTASH_CLOAKING_REDIS_REST_TOKEN
◈ Injected .env file env var: IPDATA_API_KEY
◈ Loaded function create-redirects.
◈ Loaded function delete-redirects.
◈ Loaded function landers.
◈ Loaded function offers.
◈ Loaded function query.
◈ Functions server is listening on 53716
◈ Starting Netlify Dev with Eleventy

> valhalla-landers@1.0.0 dev
> run-p clean sass webpack:dev 11ty:dev


> valhalla-landers@1.0.0 clean
> rimraf dist/*


> valhalla-landers@1.0.0 sass
> sass resources/scss/main.scss dist/assets/css/main.css


> valhalla-landers@1.0.0 webpack:dev
> webpack --mode development --watch


> valhalla-landers@1.0.0 11ty:dev
> ELEVENTY_ENV=development npx @11ty/eleventy

1 asset
236 modules
webpack 5.69.1 compiled successfully in 1460 ms
[11ty] Writing dist/robots.txt from ./site/robots.njk
[11ty] Writing dist/9ffe3f93-8774-4e12-a1bf-540baec15ebb/index.html from ./site/index.njk
[11ty] Writing dist/70ffc2c1-feb1-47ea-ac72-63b2411ee09c/index.html from ./site/index.njk
[11ty] Writing dist/63e88395-9247-4b88-92c9-1ef649944149/index.html from ./site/index.njk
[11ty] Writing dist/pages/terms-of-use/index.html from ./site/pages/terms-of-use.md (njk)
[11ty] Writing dist/pages/contact/index.html from ./site/pages/contact.md (njk)
[11ty] Writing dist/pages/privacy-policy/index.html from ./site/pages/privacy-policy.md (njk)
[11ty] Writing dist/pages/legal-disclosure/index.html from ./site/pages/legal-disclosure.md (njk)
[11ty] Writing dist/404.html from ./site/404.md (njk)
[11ty] Writing dist/pages/about/index.html from ./site/pages/about.md (njk)
[11ty] Benchmark    798ms  54%     1× (Data) `./site/_data/valhalla.js`
[11ty] Copied 5 files / Wrote 10 files in 1.02 seconds (102.0ms each, v1.0.0)
⠙ Waiting for framework port 8080. This can be configured using the 'targetPort' property in the netlify.toml
```

My `package.json` scripts & deps:
```json

  "scripts": {
    "help": "npx @11ty/eleventy --help",
    "clean": "rimraf dist/*",
    "sass": "sass resources/scss/main.scss dist/assets/css/main.css",
    "webpack:dev": "webpack --mode development --watch",
    "webpack:prod": "webpack --mode production",
    "11ty:dev": "ELEVENTY_ENV=development npx @11ty/eleventy --serve --watch",
    "11ty:prod": "ELEVENTY_ENV=production npx @11ty/eleventy",
    "dev": "run-p clean sass webpack:dev 11ty:dev",
    "prod": "run-s sass 11ty:prod webpack:prod"
  },
  "dependencies": {
    "@babel/runtime": "^7.17.2",
    "@upstash/redis": "^1.0.3",
    "alpinejs": "^3.9.1",
    "axios": "^0.27.2",
    "base64-url": "^2.3.3",
    "ipdata": "^1.1.2",
    "unfetch": "^4.2.0"
  },
  "devDependencies": {
    "@11ty/eleventy": "^1.0.0",
    "@11ty/eleventy-navigation": "^0.3.2",
    "@babel/plugin-transform-runtime": "^7.17.0",
    "@babel/preset-env": "^7.16.11",
    "@babel/runtime-corejs3": "^7.17.2",
    "babel-loader": "^8.2.3",
    "clean-css": "^5.2.4",
    "gorko": "^0.4.1",
    "html-minifier": "^4.0.0",
    "netlify-cli": "^10.11.1",
    "node-html-parser": "^4.1.0",
    "npm-run-all": "^4.1.5",
    "postcss": "^8.3.5",
    "purgecss": "^4.1.3",
    "resolve-url-loader": "^4.0.0",
    "rimraf": "^2.7.1",
    "sass": "^1.49.9",
    "webpack": "^5.69.1",
    "webpack-cli": "^4.9.2"
  }
```

### Steps to reproduce

Upgrade to latest version, use netlify.toml and package.json i posted here (I assume)

### Configuration

```toml
[build]
build = "/"
command = "npm run prod"
publish = "dist"

[dev]
command = "npm run dev"
framework = "eleventy"
publish = "dist"

[[headers]]
# Define which paths this specific [[headers]] block will cover.
for = "/*"

[headers.values]
Referrer-Policy = "no-referrer"
X-Frame-Options = "DENY"
X-Robots-Tag = "noindex"
X-XSS-Protection = "1; mode=block"

[functions]
# Directory with serverless functions, including background functions,
# to deploy. This is relative to the base directory if one has been set, 
# or the root directory if a base hasn’t been set.
directory = ".netlify/functions"
```

### Environment

❯ npx envinfo --system --binaries --npmPackages netlify-cli --npmGlobalPackages netlify-cli

  System:
    OS: macOS 12.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 919.10 MB / 32.00 GB
    Shell: 3.4.1 - /usr/local/bin/fish
  Binaries:
    Node: 18.6.0 - /usr/local/bin/node
    npm: 8.13.2 - /usr/local/bin/npm
  npmPackages:
    netlify-cli: ^10.11.1 => 10.11.1
  npmGlobalPackages:
    netlify-cli: 10.11.1

### PR DESCRIPTION
**Pull request by danez**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Updates wait-port to v1 which now tries both IPv6 and IPv4 and also reports where it found the open port. The proxy will then proxy to the correct address.

This should once and for all solve the IPv4 vs IPv6 problem with node 17+

Fixes #4845

Maybe helps with #3735 and #4942 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/commands/dev/dev.js [hunk 1]
```diff
@@ -414,6 +423,8 @@ const validateGeoCountryCode = (arg) => {
   return arg.toUpperCase()
 }
 
+const IP_VERSION_6 = 6
```
Comment (lukasholzer) [inline src/commands/dev/dev.js:426]: IMHO it's easier to read to use the 6 directly and disable the lint rule for magic numbers

Comment (danez) [reply src/commands/dev/dev.js:426]: Agree.

File: src/commands/dev/dev.js [hunk 2]
```diff
@@ -178,18 +178,25 @@ const runCommand = (command, env = {}, spinner = null) => {
   return commandProcess
 }
 
+/**
+ * @typedef StartReturnObject
+ * @property {4 | 6 | undefined=} ipVersion The version the open port was found on
```
Comment (eduardoboucas) [inline src/commands/dev/dev.js:183]: I'm assuming this is a typo?

```suggestion
 * @property {4 | 6 | undefined} ipVersion The version the open port was found on
```

File: src/commands/dev/dev.js [hunk 3]
```diff
@@ -178,18 +178,25 @@ const runCommand = (command, env = {}, spinner = null) => {
...
+ */
+
 /**
  * Start a static server if the `useStaticServer` is provided or a framework specific server
  * @param {object} config
  * @param {Partial<import('../../utils/types').ServerSettings>} config.settings
- * @returns {Promise<void>}
+ * @returns {Promise<StartReturnObject>}
```
Comment (eduardoboucas) [inline src/commands/dev/dev.js:190]: The return value is always an empty object, right?

File: src/commands/dev/dev.js [hunk 4]
```diff
@@ -221,6 +228,8 @@ const startFrameworkServer = async function ({ settings }) {
     log(NETLIFYDEVERR, `Please make sure your framework server is running on port ${settings.frameworkPort}`)
     exit(1)
   }
+
+  return { ipVersion: port?.ipVersion }
```
Comment (eduardoboucas) [inline src/commands/dev/dev.js:232]: oooh, I like that we can finally use optional chaining in the CLI!

File: src/commands/dev/dev.js [hunk 5]
```diff
@@ -501,7 +510,11 @@ const dev = async (options, command) => {
 
   log(`${NETLIFYDEVWARN} Setting up local development server`)
 
-  const devCommand = () => startFrameworkServer({ settings })
+  const devCommand = async () => {
+    const { ipVersion } = await startFrameworkServer({ settings })
+    // eslint-disable-next-line no-magic-numbers
```
Comment (eduardoboucas) [inline src/commands/dev/dev.js:515]: nit: I would actually create a constant for this.

**Review by eduardoboucas — APPROVED** [review]
Nice! I left minor comments, mostly cosmetic.

File: src/commands/dev/dev.js (hunk 2 above)
Comment (danez) [reply src/commands/dev/dev.js:183]: No the equal sign marks the field as optional

File: src/commands/dev/dev.js (hunk 3 above)
Comment (danez) [reply src/commands/dev/dev.js:190]: Either empty or has the property `ipVersion`

File: src/commands/dev/dev.js (hunk 4 above)
Comment (lukasholzer) [reply src/commands/dev/dev.js:232]: yea because we dropped node 12 🥳

File: src/commands/dev/dev.js (hunk 5 above)
Comment (lukasholzer) [reply src/commands/dev/dev.js:515]: IMHO I think it's easier to understand in the context of `6 ? '::1' : '127.0.0.1'`

But I think it's personal preference just my 2 cents

File: src/commands/dev/dev.js [hunk 6]
```diff
@@ -200,17 +207,17 @@ const startFrameworkServer = async function ({ settings }) {
...
   runCommand(settings.command, settings.env, spinner)
 
+  let port
   try {
-    const open = await waitPort({
+    port = await waitPort({
       port: settings.frameworkPort,
-      // Cannot use `localhost` as it may point to IPv4 or IPv6 depending on node version and OS
-      host: '127.0.0.1',
+      host: 'localhost',
```
Comment (benedfit) [inline src/commands/dev/dev.js:214]: @danez @eduardoboucas this has caused a regressions for one of our own CI pipelines https://netlify.slack.com/archives/CPJQSPQE4/p1665509398672569

Could we use `settings.frameworkHost` here to solve this?

**Pull request merged**